spec/02-integration/01-helpers/01-helpers_spec.lua

```diff
@@ -61,6 +61,23 @@ for _, strategy in helpers.each_strategy() do
         assert.equals("bob",    params["names[2]"])
         assert.equals("casius", params["names[3]"])
       end)
+
+      it("encodes nested tables and arrays in a way Lapis-compatible way when using form-urlencoded content-type", function()
```

Comment:
Note that we decode our Admin API request bodies [ourselves](https://github.com/Kong/kong/blob/master/kong/api/init.lua#L38) (and https://github.com/Kong/kong/blob/master/kong/api/api_helpers.lua#L23), and do not rely on Lapis for it. We should write unit tests asserting that the those decoders can indeed parse content encoded with nested tables. 

Also, note the typo (repeated "way") in this test's title.

Reply:
It turns out our api_helper was not able to handle "full" nested structures, only a subset. I have added tests for this and the changes needed in a separate commit. I also removed the "way way" typo.